FWCore/ParameterSet/test/test_catch2_ps.cc

```diff
@@ -0,0 +1,875 @@
+    testbody<bool>(true);
+  }
+
+  SECTION("int") {
+    testbody<int>(-std::numeric_limits<int>::max());
+    testbody<int>(-2112);
+    testbody<int>(-0);
+    testbody<int>(0);
+    testbody<int>(35621);
+    testbody<int>(std::numeric_limits<int>::max());
+  }
+
+  SECTION("uint") {
+    testbody<unsigned int>(0);
+    testbody<unsigned int>(35621);
+    testbody<unsigned int>(std::numeric_limits<unsigned int>::max());
+
+    testbody<std::vector<unsigned int>>(std::vector<unsigned int>());
+    testbody<std::vector<unsigned int>>(std::vector<unsigned int>(1, 35621));
+    testbody<std::vector<unsigned int>>(std::vector<unsigned int>(1, std::numeric_limits<unsigned int>::max()));
+  }
+
+  SECTION("double") {
+    testbody<double>(-1.25);
+    testbody<double>(-0.0);
+    testbody<double>(0.0);
+    testbody<double>(1.25);
+    //testbody<double>(1.0/0.0);  // parameter set does not handle infinity?
+    //testbody<double>(0.0/0.0);  // parameter set does not handle NaN?
+    testbody<double>(-2.3456789e-231);
+    testbody<double>(-2.3456789e231);
+    testbody<double>(2.3456789e-231);
+    testbody<double>(2.3456789e231);
+    double oneThird = 1.0 / 3.0;
+    testbody<double>(oneThird);
+  }
+
+  SECTION("edm::decode_vstring_extent") {
+    SECTION("empty") {
+      auto ret = edm::decode_vstring_extent("{}");
+      REQUIRE(ret);
+      CHECK(*ret == "{}");
+    }
+    SECTION("empty stuff after") {
+      auto ret = edm::decode_vstring_extent("{}xava");
+      REQUIRE(ret);
+      CHECK(*ret == "{}");
+    }
+    SECTION("one empty string") {
+      auto ret = edm::decode_vstring_extent(std::string_view("{\0\0}", 4));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view("{\0\0}", 4));
+    }
+    SECTION("one empty string, stuff after") {
+      auto ret = edm::decode_vstring_extent(std::string_view("{\0\0})aa", 6));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view("{\0\0}", 4));
+    }
+    SECTION("simple multi-entry") {
+      auto ret = edm::decode_vstring_extent(std::string_view("{\0one\0,1\0,TWO\0,three\0}", 22));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view(std::string_view("{\0one\0,1\0,TWO\0,three\0}", 22)));
+    }
+    SECTION("simple multi-entry, stuff after") {
+      auto ret = edm::decode_vstring_extent(std::string_view("{\0one\0,1\0,TWO\0,three\0});", 24));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view(std::string_view("{\0one\0,1\0,TWO\0,three\0}", 22)));
+    }
+  }
+  SECTION("edm::decode_string_extent") {
+    SECTION("empty") {
+      auto ret = edm::decode_string_extent(std::string_view("\0", 1));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view("\0", 1));
+    }
+    SECTION("empty stuff after") {
+      auto ret = edm::decode_string_extent(std::string_view("\0xava", 5));
+      REQUIRE(ret);
+      CHECK(*ret == std::string_view("\0", 1));
+    }
+    SECTION("string with null") {
+      auto ret = edm::decode_string_extent(std::string_view("\0\0\0", 3));
```

Comment:
Would it be useful to test "string with null" also with `decode_vstring_extent()`?

Reply:
done